src/main.rs

```diff
@@ -33,33 +34,77 @@ struct Args {
     wikipedia_urls: Option<PathBuf>,
 }
 
-fn write(dir: impl AsRef<Path>, page: Page) -> anyhow::Result<()> {
-    let Some(qid) = page.main_entity.map(|e| e.identifier) else {
-        // TODO: handle and still write
-        bail!("Page in list but without wikidata qid: {:?} ({})", page.name, page.url);
+/// Determine the directory to write the article contents to, create it, and create any necessary symlinks to it.
+fn create_article_dir(base: impl AsRef<Path>, page: &Page) -> anyhow::Result<PathBuf> {
+    let Some(qid) = page.wikidata() else {
+        // TODO: Handle and write to wikipedia title directory
+        info!("Page without wikidata qid: {:?} ({})", page.name, page.url);
+        return Ok(Default::default());
     };
 
-    let mut filename = dir.as_ref().to_owned();
-    filename.push(qid);
+    // Otherwise use wikidata as main directory and symlink from wikipedia titles.
+
+    let wikidata_dir = qid.get_dir(base.as_ref().to_owned());
+    create_dir_all(&wikidata_dir)
+        .with_context(|| format!("creating wikidata directory {:?}", &wikidata_dir))?;
+
+    // Write symlinks to main directory.
+    // TODO: Only write redirects that we care about.
+    for title in page.titles() {
+        let title = match title {
+            Err(e) => {
+                warn!("Unable to parse title for page {:?}:  {:#}", page.name, e);
+                continue;
+            }
+            Ok(title) => title,
+        };
+
+        // NOTE: Some wikipedia titles have '/' in them.
+        let wikipedia_dir = title.get_dir(base.as_ref().to_owned());
+        // TODO: handle incorrect links, directories
```

Comment:
For example?

Reply:
The "incorrect links, directories" refers to updating a directory tree from a previous run, instead of starting from scratch. Right now the behavior is to skip any file that exists.